web/src/views/environments/EnvironmentCard.jsx

```diff
@@ -0,0 +1,17 @@
+const EnvironmentCard = ({ envs }) => {
+    return (
+        <div className="px-4 py-8 sm:px-0">
+            {Object.keys(envs).map(env => (
+                <div className='bg-white overflow-hidden shadow rounded-lg my-4 w-fullpx-4 py-5 sm:px-6 focus:outline-none'>
```

Comment:
I think it's better if you factor the card into a component, not all cards in one.

Right now the two approaches are actually mixed. You call the component `EnvironmentCard` while it holds multiple cards.

Also it is better if the containing div and the control logic is one unit, and the card content is another is another logical unit. You either change the control logic, or the stuff inside.

```
 <div className="px-4 py-8 sm:px-0">
            {Object.keys(envs).map(env => (...
)}
</div>
```